src/Aspire.Hosting.Kafka/KafkaBuilderExtensions.cs

```diff
@@ -110,19 +110,16 @@ public static IResourceBuilder<KafkaServerResource> WithKafkaUI(this IResourceBu
                 .WithHttpEndpoint(targetPort: KafkaUIPort)
                 .ExcludeFromManifest();
 
-            builder.ApplicationBuilder.Eventing.Subscribe<AfterEndpointsAllocatedEvent>((e, ct) =>
+            builder.ApplicationBuilder.Eventing.Subscribe<BeforeResourceStartedEvent>(kafkaUi, (e, ct) =>
             {
                 var kafkaResources = builder.ApplicationBuilder.Resources.OfType<KafkaServerResource>();
 
                 int i = 0;
                 foreach (var kafkaResource in kafkaResources)
                 {
-                    if (kafkaResource.InternalEndpoint.IsAllocated)
```

Comment:
For container-to-container, we don't need to wait for the endpoint to be allocated.

Reply:
Reverted these particular changes.